lms/views/api/canvas_studio.py

```diff
@@ -135,15 +148,31 @@ def via_url(request):
     document_url = assignment.document_url
     media_id = CanvasStudioService.media_id_from_url(document_url)
     if not media_id:
-        raise HTTPBadRequest("Unable to get Canvas Studio media ID")
+        raise CanvasStudioLaunchError(
+            "canvas_studio_media_not_found", "Unable to get Canvas Studio media ID"
+        )
```

Comment:
This particular error should never happen unless the document URL got edited on Canvas's side or we screwed up when generating it. However there is another related error which could happen, which is if the video gets removed in Canvas. I've used an error code which reflects that more general case.